src/group_theory/subgroup/basic.lean

```diff
@@ -1416,6 +1419,32 @@ lemma normalizer_condition.normal_of_coatom
   (hnc : normalizer_condition G) (hmax : is_coatom H) : H.normal :=
 normalizer_eq_top.mp (hmax.2 _ (hnc H (lt_top_iff_ne_top.mpr hmax.1)))
 
+end normalizer
+
+section centralizer
+
+/-- The `centralizer` of `H` is the subgroup of `g : G` commuting with every `h : H`. -/
+@[to_additive "The `centralizer` of `H` is the additive subgroup of `g : G` commuting with
+every `h : H`."]
+def centralizer : subgroup G :=
+{ carrier := {g : G | ∀ h ∈ H, h * g = g * h},
```

Comment:
```suggestion
{ carrier := set.centralizer H,
```
Or similar

Reply:
Are you suggesting that I define `set.centralizer` (or perhaps `set_centralizer` to match `set_normalizer`)? In that case, the definition of centralizer could become one-line, I think.

Reply:
Although one issue with defining `centralizer` in terms of `set.centralizer` is that you will end up with a bunch of coercions from subgroups to sets, which is annoying. I presume that's why `normalizer` is not defined in terms of `set_normalizer`.

Reply:
My understanding was that the set version already exists

Reply:
Perhaps it was removed by accident

Reply:
I see. Well, I can add the set-version if you want, but I'm not sure about defining `centralizer` in terms of the set-version, due to the coercions showing up.

Reply:
You can avoid the coercions if we instead have
```lean
protected def set.centralizer (p : S → Prop) : set S := { x | ∀ a, p a → commute a x } 
```
It seems there was quite a lot of theory about `set.centralizer` that just got deleted - it would maybe make sense to just restore all the lost work first, then apply your changes.